src/components/TelemetryModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {useState} from 'react';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+import Cookies from 'universal-cookie';
+import {close, getCurrentHub} from '@sentry/browser';
+import moment from 'moment';
+
+/**
+ * Note this file is covered by Manual Test TI.2
+ */
+
+/**
+ * The Telemetry Popup which asks for user consent to collect
+ * telemetry.
+ * Pre-Conditions: The user has not already set the telemetry
+ * cookie.
+ * Post-Conditions: The telemetry cookie is set in accordance
+ * with the users selection.
+ * @return {any} A react component which is the telemetry
+ * modal rendered
+ */
+export default function TelemetryModal() {
+  const cookies = new Cookies();
+  const prompt = cookies.get('telemetry');
+
+  // The assertion is handled here to check if the cookie has been set.
+  const [show, setShow] = useState(prompt === undefined);
```

Comment:
is this the right syntax? I thought assertions used the word 'assert'

Reply:
Heya, perhaps assertion isn't the right word. And maybe it's not a pre-condition. It's sort of an inbetween. Basically this dialog only appears if you don't have the cookie already, but the function checks that here. It's not necessarily an error state if the cookie exists, it just mean we don't need to render the popup.